src/main.js

```diff
@@ -1,45 +1,79 @@
 /* eslint-disable eqeqeq */
 /* eslint-disable import/no-cycle */
-import { todoItemsList, todos, addToLocalStorage } from './index.js';
-
-export function renderTodos(todos) {
-  // clear everything inside <ul> with class=todo-items
-  todoItemsList.innerHTML = '';
-  todos.forEach((item) => {
-    // check if the item is completed
-    const checked = item.completed ? 'checked' : null;
-
-    const li = document.createElement('li');
-    // <li class="item"> </li>
-    li.setAttribute('class', 'item');
-    // <li class="item" data-key="20200708"> </li>
-    li.setAttribute('data-key', item.id);
-    /* <li class="item" data-key="20200708">
-          <input type="checkbox" class="checkbox">
-          <span>Go to Gym </span>
-          <button class="delete-button">X</button>
-        </li> */
-    if (item.completed === true) {
-      li.classList.add('checked');
+import {
+  todoItemsList, todos, todoInput, updateLocalStorage,
+} from './index.js';
+
+class App {
+  static addTodo(item) {
+    if (item !== '') {
+      const todo = {
+        id: Date.now(),
+        description: item,
+        completed: false,
+        index: todos.length + 1,
+      };
+
+      todos.push(todo);
+      updateLocalStorage(todos);
+      todoInput.value = '';
     }
+  }
+
+  static renderTodos(todos) {
+    todoItemsList.innerHTML = '';
+    todos.forEach((item) => {
+      // check if the item is completed
+      const checked = item.completed ? 'checked' : null;
+
+      const li = document.createElement('li');
+      li.setAttribute('class', 'item');
+      // <li class="item" data-key="20200708"> </li>
+      li.setAttribute('data-key', item.id);
+
+      if (item.completed === true) {
+        li.classList.add('checked');
+      }
+
+      li.innerHTML = `
+        <input type="checkbox" class="checkbox" ${checked}>
+        <input type="text" value="${item.description}" class="input-desc ${checked}">
+        <button class="delete-button">X</button>
+      `;
+      todoItemsList.append(li);
+    });
+  }
 
-    li.innerHTML = `
-      <input type="checkbox" class="checkbox" ${checked}>
-      <span>${item.description}</span>
-      <button class="delete-button">X</button>
-    `;
-    todoItemsList.append(li);
-  });
+  static toggle(id) {
+    todos.forEach((item) => {
+      // use == not ===, because here types are different. One is number and other is string
+      if (item.id == id) item.completed = !item.completed; // toggle the value
+    });
+    updateLocalStorage(todos);
+  }
+
+  static updateItem(newValue, id) {
+    todos.forEach((list) => {
+      if (list.id == id) list.description = newValue;
+    });
+    updateLocalStorage(todos);
+  }
+
+  static deleteTodo(id) {
+    const ind = todos.findIndex((obj) => obj.id == id);
+    todos.splice(ind, 1);
+    updateLocalStorage(todos);
+  }
```

Comment:
- [ ] Kindly make sure when an item is deleted, the list of todos is resorted correctly by index.
For example, we have one, two, and three todo items in the list, one has an index = 0, two = 1, and three = 2, and we deleted number two on the list (index = 1), the 3rd item should have an index = 1.
>Please go ahead and check out the project requirements for this point right [here](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m3_add_remove.md#:~:text=Deleting%20a%20task%20should,saved%20in%20local%20storage.).

Reply:
Hello @MahmoudBakr23  Thank You for the feedback!, I really appreciated, however I do have a `function updateIndex() ` in my **index.js** file and I do call it inside the `function updateLocalStorage` it's job is to change the index value inside the to-do list or the item object. And I have tested it, it's working on my side.

Thank You Again for the review.

Reply:
Thank you for communicating with me ❤️